tests/E2E/RoleTest.php

```diff
@@ -0,0 +1,137 @@
+<?php
+
+/**
+ * This file is part of the contentful-management.php package.
+ *
+ * @copyright 2015-2017 Contentful GmbH
+ * @license   MIT
+ */
+
+namespace Contentful\Tests\E2E;
+
+use Contentful\Management\Query;
+use Contentful\Management\Resource\Role;
+use Contentful\Management\Role\Constraint\AndConstraint;
+
+use Contentful\Management\Role\Constraint\EqualityConstraint;
+use Contentful\Management\Role\Constraint\NotConstraint;
+use Contentful\Management\Role\Constraint\OrConstraint;
+use Contentful\Management\Role\Permissions;
+use Contentful\Management\Role\Policy;
+use Contentful\Tests\End2EndTestCase;
+
+class RoleTest extends End2EndTestCase
+{
+    /**
+     * @vcr e2e_role_get_one.json
+     */
+    public function testGetRole()
+    {
+        $manager = $this->getReadWriteSpaceManager();
+
+        $role = $manager->getRole('6khUMmsfVslYd7tRcThTgE');
+
+        $this->assertEquals('Developer', $role->getName());
+        $this->assertEquals('Allows reading Entries and managing API Keys', $role->getDescription());
+
+        $policies = $role->getPolicies();
+        $this->assertCount(2, $policies);
+        $this->assertEquals('allow', $policies[0]->getEffect());
+        $this->assertEquals(['read'], $policies[0]->getActions());
+        $this->assertInstanceOf(AndConstraint::class, $policies[0]->getConstraint());
+        $constraint = $policies[0]->getConstraint()->getChildren()[0];
+        $this->assertInstanceOf(EqualityConstraint::class, $constraint);
+        $this->assertEquals('sys.type', $constraint->getDoc());
+        $this->assertEquals('Entry', $constraint->getValue());
+
+        $permissions = $role->getPermissions();
+        $this->assertEquals('all', $permissions->getContentDelivery());
+        $this->assertEquals(['read'], $permissions->getContentModel());
+        $this->assertEquals([], $permissions->getSettings());
+    }
+
+    /**
+     * @vcr e2e_role_get_collection.json
+     */
+    public function testGetRoles()
+    {
+        $manager = $this->getReadWriteSpaceManager();
+        $roles = $manager->getRoles();
+
+        $this->assertInstanceOf(Role::class, $roles[0]);
+
+        $query = (new Query)
+            ->setLimit(1);
+        $roles = $manager->getRoles($query);
+        $this->assertInstanceOf(Role::class, $roles[0]);
+        $this->assertCount(1, $roles);
```

Comment:
Should you also assert on whether the content of the roles array is the same, not just the count (aka can you check if the values set on the first entry are the ones you would expect?)